source/extensions/http/header_validators/envoy_default/http2_header_validator.cc

```diff
@@ -116,7 +115,14 @@ Http2HeaderValidator::validateRequestHeaderMap(::Envoy::Http::RequestHeaderMap&
             UhvResponseCodeDetail::get().InvalidMethod};
   }
 
-  auto is_connect_method = header_map.method() == header_values_.MethodValues.Connect;
+  // The CONNECT method with the ":protocol" header is called the extended CONNECT and covered in
+  // https://datatracker.ietf.org/doc/html/rfc8441#section-4
+  // For the purposes of header validation the extended CONNECT is treated as generic (non CONNECT)
+  // HTTP/2 requests.
+  auto is_connect_method = header_map.method() == header_values_.MethodValues.Connect &&
+                           header_map.getProtocolValue().empty();
+  auto is_extended_connect_method = header_map.method() == header_values_.MethodValues.Connect &&
```

Comment:
hasn't the method been normalized by the point we hit the validator?

Reply:
also either way can we normalize from is_connect_method to is_standard_connect_request and is_extended_connect_request since it's really not about the method?

Reply:
With UHV transformation from extended CONECT to upgrade will happen after header map was validated.